src/index.ts

```diff
@@ -1,74 +1,100 @@
 
   public options: ElmPluginOptions;
 
   constructor(options: ElmPluginOptions = {}) {
-    this.options = Object.assign({}, options);
+    this.options = { ...options };
   }
 
-  public init(context: WorkFlowContext) {
+  public init(context: WorkFlowContext): void {
     context.allowExtension(".elm");
   }
 
-  public transform(file: File) {
-    file.loadContents();
+  public getElmMakePath(): string {
+    try {
+      return resolve("node_modules/.bin/elm-make");
+    } catch (_) {}
 
-    // Get the path to elm-make
-    const elmMakePath = getElmMakePath();
-
-    return tmp()
-      .then(tmpFilename => tmpFilename + ".js")
-      .then(tmpFilename => new Promise((resolve, reject) => {
-        // Construct the arguments for elm-make
-        const args = [
-          "--yes",
-          "--output",
-          tmpFilename,
-          this.options.warn ? "--warn" : null,
-          this.options.debug ? "--debug" : null,
-          file.absPath
-        ].filter(x => x !== null);
-
-        const proc = spawn(elmMakePath, args, { stdio: "inherit" });
-
-        proc.on("close", code => {
-          if (code === 0) {
-            fs.readFile(tmpFilename, (err, data) => {
-              if (err) reject(err);
-              file.contents = data.toString();
-              resolve(file);
-            });
-          } else {
-            reject("Failed to compile Elm");
-          }
-        });
-      }));
+    return "elm-make";
   }
 
-}
+  public handleError(elmMakePath: string, err: NodeJS.ErrnoException): void {
+    if (err.code === "ENOENT") {
+      console.error(
+        `Could not find Elm compiler @ "${elmMakePath}"
+         \n Is it installed? If not, please install "elm" via npm`
+      );
+    } else if (err.code === "EACCES") {
+      console.error(
+        `Elm compiler @ "${elmMakePath}" did not have permission to run. 
+        You may need give it executable permissions.`
+      );
+    } else {
+      console.error(
+        `Error attempting to run Elm compiler @ "${elmMakePath}" \n ${err}`
+      );
+    }
+  }
 
-function getElmMakePath() {
-  try {
-    return path.resolve("node_modules/.bin/elm-make");
-  } catch (_) { }
+  public async transform(file: File): Promise<any> {
+    file.loadContents();
 
-  return "elm-make";
+    // Get the path to elm-make
+    const elmMakePath: string = this.getElmMakePath();
+
+    // Create temporary JS file
+    const tmpFilename: string = `${await tmp()}.js`;
+
+    return new Promise((resolve, reject) => {
+      // Construct the arguments for elm-make
+      const args = [
+        "--yes",
+        "--output",
+        tmpFilename,
+        this.options.warn ? "--warn" : null,
+        this.options.debug ? "--debug" : null,
+        file.absPath
+      ].filter(x => x !== null);
+
+      const proc: ChildProcess = spawn(elmMakePath, args, { stdio: "inherit" });
+
+      proc.on("error", (err: NodeJS.ErrnoException) =>
+        this.handleError(elmMakePath, err)
```

Comment:
This needs to reject the main Promise as well as writing the error to `console.error`.  This will be a bit fiddly to do with `handleError` as a separate function, so I would just leave its body in here.